core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java

```diff
@@ -460,6 +483,29 @@ public boolean namespaceExists(String namespace) throws ExecutionException {
     return databaseExists(namespace);
   }
 
+  @Override
+  public void repairTable(
+      String namespace, String table, TableMetadata metadata, Map<String, String> options)
+      throws ExecutionException {
+    try {
+      // Since the metadata table may be missing, we cannot use CosmosAdmin.tableExists() as it
+      // query the metadata table to verify if the given table exists
+      client.getDatabase(namespace).getContainer(table).read();
+    } catch (CosmosException e) {
+      if (e.getStatusCode() == 404) {
+        throw new IllegalArgumentException(
+            "The table " + getFullTableName(namespace, table) + "  does not exist");
+      }
+    }
+    createMetadataDatabaseAndContainerIfNotExists();
+    if (getTableMetadata(namespace, table) == null) {
+      putTableMetadata(namespace, table, metadata);
+    }
+    if (!storedProcedureExists(namespace, table)) {
+      addStoredProcedure(namespace, table);
```

Comment:
Hmm, this is a little exceptional since stored procedures are not table metadata.

Reply:
I will answer here for both of your comments. 
 
Our goal in creating this method was to address two needs : 
- Be able to repair table metadata which is common to all storage
- Be able to restore a table stored procedure which is Cosmos DB specific

Adding a separate method to the Admin interface to address the stored procedure repair does not fit the Admin interface philosophy so far. For example, specific storage options to configure table (like Dynamo DB RU or disabling Dynamo DB auto-backup)  are addressed in the `createTable` method using the `Map<String, String> options` parameter. We decided to have a similar approach for the `createTable` method.
That’s why the repairTable naming is quite vague, it indeed repairs metadata but might also repair other things.